src/Dapr.Client/PublishHttpClient.cs

```diff
@@ -0,0 +1,95 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+namespace Dapr
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Text.Json;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using static Dapr.DaprUris;
+
+    /// <summary>
+    /// A client for interacting with the Dapr publish endpoints using <see cref="HttpClient" />.
+    /// </summary>
+    public class PublishHttpClient : PublishClient
+    {
+        private readonly HttpClient client;
+        private readonly JsonSerializerOptions serializerOptions;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PublishHttpClient"/> class.
+        /// </summary>
+        /// <param name="client">The <see cref="HttpClient" />.</param>
+        /// <param name="serializerOptions">The <see cref="JsonSerializerOptions" />.</param>
+        public PublishHttpClient(HttpClient client, JsonSerializerOptions serializerOptions = null)
+        {
+            if (client is null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            this.client = client;
+            this.serializerOptions = serializerOptions;
+        }
+
+        /// <inheritdoc/>
+        public override async Task PublishEventAsync<TRequest>(string topicName, TRequest publishContent, CancellationToken cancellationToken = default)
+        {
+            if (string.IsNullOrEmpty(topicName))
+            {
+                throw new ArgumentNullException("The value cannot be null or empty", nameof(topicName));
+            }
+
+            if (publishContent is null)
+            {
+                throw new ArgumentNullException("The value cannot be null or empty", nameof(publishContent));
+            }
+
+            await this.MakePublishHttpRequest(topicName, JsonSerializer.Serialize(publishContent), cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        public override async Task PublishEventAsync(string topicName, CancellationToken cancellationToken = default)
+        {
+            if (string.IsNullOrEmpty(topicName))
+            {
+                throw new ArgumentNullException("The value cannot be null or empty", nameof(topicName));
+            }
+
+            await this.MakePublishHttpRequest(topicName, string.Empty, cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<HttpResponseMessage> MakePublishHttpRequest(string topicName, string publishContents, CancellationToken cancellationToken)
```

Comment:
If you want to make this slightly more efficient, pull [this](https://github.com/dapr/dotnet-sdk/blob/b02aa3f02ba3084b480f668c28c5d5d79b89c7cd/src/Dapr.Client/StateHttpClient.cs#L173) out into an internal class and you can use it here.